src/widgets/helper/color/ColorButton.cpp

```diff
@@ -0,0 +1,76 @@
+#include "widgets/helper/color/ColorButton.hpp"
+
+#include "widgets/helper/color/Checkerboard.hpp"
+
+#include <QPainterPath>
+
+namespace chatterino {
+
+ColorButton::ColorButton(QColor color, QWidget *parent)
+    : QAbstractButton(parent)
+    , currentColor_(color)
+{
+    this->setSizePolicy({QSizePolicy::Expanding, QSizePolicy::Expanding});
+    this->setMinimumSize({30, 30});
+}
+
+QSize ColorButton::sizeHint() const
+{
+    return {50, 30};
+}
+
+void ColorButton::setColor(const QColor &color)
+{
+    this->currentColor_ = color;
+    this->update();
```

Comment:
Should we check to only call update if currentColor_ and color are not the same? or is that overengineering this use case

Reply:
Yes we should check this.